securityhub-iam-password-policy-*

jihygk opened the pull request:
Adds remediations for securityhub-iam-password-* rules

securityhub-iam-password-policy-ensure-expires
securityhub-iam-password-policy-lowercase-letter-check
securityhub-iam-password-policy-minimum-length-check
securityhub-iam-password-policy-number-check
securityhub-iam-password-policy-prevent-reuse-check
securityhub-iam-password-policy-symbol-check

Commit:
add .idea/ to .gitignore

Commit:
add iam-password-policy remediations

mlevit commented on auto-remediate/lambda_handler.py:
```diff
@@ -56,6 +56,8 @@ def remediate(self):
                             remediation = self.security_hub.s3_bucket_public_read_prohibited(config_rule_resource_id)
                         elif 's3-bucket-public-write-prohibited' in config_rule_name:
                             remediation = self.security_hub.s3_bucket_public_write_prohibited(config_rule_resource_id)
+                        elif 'iam-password-policy' in config_rule_name:
```
@jihygk can we maybe these in alphabetical order

jihygk replied:
8af426d

mlevit commented on auto-remediate/security_hub_rules.py:
```diff
@@ -5,7 +5,40 @@
         self.logging = logging
-    
+
+    def apply_cis_password_policy(self, resource_id):
```
@jihygk can we rename this to `iam_password_policy` to match the Config Rule names

jihygk replied:
1afb192

Commit:
fix typo

mlevit commented on auto-remediate/security_hub_rules.py:
```diff
@@ -5,7 +5,40 @@
+            self.logging.info("Updated IAM password policy with CIS AWS Foundations requirements.")
+            return True
+        except:
+            self.logging.error("Could not update IAM password policy for {resource}.".format(resource=resource_id))
```
@jihygk Could you change this keep it consistent:

`self.logging.error("Could not update IAM password policy for '%s'." % resource_id))`

jihygk replied:
I'll refactor existing stuff to `.format`

[The `%` style is not recommended.](https://docs.python.org/3/library/stdtypes.html#printf-style-string-formatting)

jihygk replied:
Actually I refactored to [PEP 498 f-strings](https://www.python.org/dev/peps/pep-0498/) which is much better in terms of readability + performance than `str.format()` or %

Commit:
change function name for consistency

Commit:
alphabetically order remediations for securityhub

Commit:
change all % strings to PEP 498 f-strings

This is a more modern way to interpolate strings; str.format() can be
quite verbose and the % method is not recommended.

mlevit reviewed (commented):
Much better man. Didn't know about this method but it's super clean and easy to read.

mlevit reviewed (commented):
Once you've done your testing, please ensure you update the README with the new Config Rules you've added support for.

Commit:
add iam:UpdateAccountPasswordPolicy to serverless.yml

Commit:
reorder iamRoleStatements in serverless.yml

mlevit commented on auto-remediate/security_hub_rules.py:
```diff
@@ -5,7 +5,40 @@
+
+        # TODO: better exception handling
+        try:
+            response = client.update_account_password_policy(
```
@jihygk fix this so it stops spamming us please :P

jihygk replied:
2488491b928edf88b5de6f7eaca7f93fcfbddd4f

Commit:
add new securityhub-iam-password-policy remediations to README

Commit:
remove assignment of IAM client response
